db/view/view.hh

```diff
@@ -123,6 +123,70 @@ bool matches_view_filter(const schema& base, const view_info& view, const partit
 
 bool clustering_prefix_matches(const schema& base, const partition_key& key, const clustering_key_prefix& ck);
 
+/**
+ * The liveness of rows resulting from an update to a table with an index over collection column
+ * depends on the liveness of the collection cells. bytes_with_action is used to return this information
+ * from the function computing the result for a particular computed column.
+ */
+struct bytes_with_action {
+    struct no_action {};
+    struct shadowable_tombstone_tag {
+        api::timestamp_type ts;
+        shadowable_tombstone into_shadowable_tombstone(gc_clock::time_point now) const {
+            return shadowable_tombstone{ts, now};
+        }
+    };
+
+    struct apply_visitor {
```

Comment:
I'm not entirely sure why this struct is necessary at all. If it only exists in order to provide `apply_tombstone`, `apply_row_marker` and `apply_all` helpers above, perhaps it could be removed altogether. More info in other comments of this review session